packages/KSelect/KSelect.vue

```diff
@@ -412,6 +459,10 @@ export default {
 
         return false
       }
+    },
+    onKeywordChange (val) {
+      this.filterStr = val
+      this.$emit('keyword-change', val)
```

Comment:
issue: we need to add a `debounce` function here. We have the `useDebounce` util function inside `/packages/utils` that you may be able to use; however, if this requires importing the `@vue/composition-api` we would rather just write a custom function.

I think setting the `debounce` value to something like `400` is reasonable; however, we could also add this as a prop.

Also, if we rename the Event emit to `query-change` then we should update this function name to `onQueryChange`

Reply:
Yes, in most cases the actual data fetching function should be debounced. But is it more suitable for autosuggest's consumer to take care of this?

For autosuggest itself, I think it's best for the `query-change` event to fire immediately after the query changes, because in some scenarios the consumer doesn't need to debounce their handlers.

For example, in both KM and Konnect we have autosuggests that do not send API requests when query changes. Instead they just filter the suggestions locally with the latest query (the suggestions are already fetched when focusing the autosuggest, like I mentioned in the previous comment). Deboucing the handler is not necessary, since no requests are sent.

The `vue-autosuggest` we currently use does not seem to have built-in debouce either, and I see Konnect handling this using `composables.useDebounce`. To me this looks like a common practice: the child component emits `change` events with every change, and the consumer component takes care of debouncing its handler.

Reply:
> But is it more suitable for autosuggest's consumer to take care of this?
> 
> For autosuggest itself, I think it's best for the `query-change` event to fire immediately after the query changes, because in some scenarios the consumer doesn't need to debounce their handlers.

Ok thanks for providing your thoughts. 

What if we just add a note in the docs page that tells the user something like, "You may want to implement a debounce to the `@query-change` event handler" and provide a simple example?

```
::: tip Note
Note text goes here... 
:::

Reply:
Sure, I'll update the docs.

Reply:
Updated in https://github.com/Kong/kongponents/pull/717/commits/651d924800c87a571f99dfe4caeaf4830a07a3b2